x/evm/keeper/state_transition.go

```diff
@@ -276,8 +273,10 @@ func (k *Keeper) ApplyTransaction(ctx sdk.Context, tx *ethtypes.Transaction) (*t
 		} else if commit != nil {
 			// PostTxProcessing is successful, commit the tmpCtx
 			commit()
-			// Since the post processing can alter the log, we need to update the result
+			// Since the post processing can alter the log and the gas used, we need
+			// to update the result
 			res.Logs = types.NewLogsFromEth(receipt.Logs)
+			res.GasUsed = receipt.GasUsed
```

Comment:
@facs95 @yihuang if the gas used from the receipt is updated in the hook, the gas used was still the old one defined by the initial result

Reply:
I see, so now hooks can increase the gas used, cool.

Reply:
the main issue is that the gas used in the hooks is not considered in the transaction. Gas estimation would also need to be updated to reflect this change.

Not sure if it's worth it...

Reply:
yeah, precompiles has an `EstimatedGas` method though, which is better in this regard.

Reply:
@fedekunze increasing the gasUsed by how much gas was used during Post-Processing is out of scope of this PR and should be treated as an extra ticket.

Reply:
yup, agree